src/plugins/marker_manager/MarkerManager.cc

```diff
@@ -0,0 +1,672 @@
+
+  material->SetAmbient(
+      _msg.material().ambient().r(),
+      _msg.material().ambient().g(),
+      _msg.material().ambient().b(),
+      _msg.material().ambient().a());
+
+  material->SetDiffuse(
+      _msg.material().diffuse().r(),
+      _msg.material().diffuse().g(),
+      _msg.material().diffuse().b(),
+      _msg.material().diffuse().a());
+
+  material->SetSpecular(
+      _msg.material().specular().r(),
+      _msg.material().specular().g(),
+      _msg.material().specular().b(),
+      _msg.material().specular().a());
+
+  material->SetEmissive(
+      _msg.material().emissive().r(),
+      _msg.material().emissive().g(),
+      _msg.material().emissive().b(),
+      _msg.material().emissive().a());
+
+  material->SetLightingEnabled(_msg.material().lighting());
+
+  return material;
+}
+
+/////////////////////////////////////////////////
+ignition::rendering::MarkerType MarkerManagerPrivate::MsgToType(
+                          const ignition::msgs::Marker &_msg)
+{
+  ignition::msgs::Marker_Type marker = this->msg.type();
+  if (marker != _msg.type() && _msg.type() != ignition::msgs::Marker::NONE)
+  {
+    marker = _msg.type();
+    this->msg.set_type(_msg.type());
+  }
+  switch (marker)
+  {
+    case ignition::msgs::Marker::BOX:
+      return ignition::rendering::MarkerType::MT_BOX;
+    case ignition::msgs::Marker::CAPSULE:
+      return ignition::rendering::MarkerType::MT_CAPSULE;
+    case ignition::msgs::Marker::CYLINDER:
+      return ignition::rendering::MarkerType::MT_CYLINDER;
+    case ignition::msgs::Marker::LINE_STRIP:
+      return ignition::rendering::MarkerType::MT_LINE_STRIP;
+    case ignition::msgs::Marker::LINE_LIST:
+      return ignition::rendering::MarkerType::MT_LINE_LIST;
+    case ignition::msgs::Marker::POINTS:
+      return ignition::rendering::MarkerType::MT_POINTS;
+    case ignition::msgs::Marker::SPHERE:
+      return ignition::rendering::MarkerType::MT_SPHERE;
+    case ignition::msgs::Marker::TEXT:
+      return ignition::rendering::MarkerType::MT_TEXT;
+    case ignition::msgs::Marker::TRIANGLE_FAN:
+      return ignition::rendering::MarkerType::MT_TRIANGLE_FAN;
+    case ignition::msgs::Marker::TRIANGLE_LIST:
+      return ignition::rendering::MarkerType::MT_TRIANGLE_LIST;
+    case ignition::msgs::Marker::TRIANGLE_STRIP:
+      return ignition::rendering::MarkerType::MT_TRIANGLE_STRIP;
+    default:
+      ignerr << "Unable to create marker of type[" << _msg.type() << "]\n";
+      break;
+  }
+  return ignition::rendering::MarkerType::MT_NONE;
+}
+
+/////////////////////////////////////////////////
+void MarkerManagerPrivate::OnWorldStatsMsg(
+  const ignition::msgs::WorldStatistics &_msg)
+{
+  std::chrono::steady_clock::duration timePoint;
+  if (_msg.has_sim_time())
+  {
+    timePoint = math::secNsecToDuration(
+        _msg.sim_time().sec(),
+        _msg.sim_time().nsec());
+    this->simTime = timePoint;
+  }
+
+  if (_msg.has_real_time())
```

Comment:
When the message has both sim and real time, the current logic will always use real time. I think we should default to sim time, because that's what users usually care about. If we want, we could add an option for users to configure whether they want to use real time.